src/Parser/Warlock/Demonology/Modules/SoulShards/SoulShardTracker.js

```diff
@@ -60,22 +48,16 @@ class SoulShardTracker extends Analyzer {
   spent = {
```

Comment:
If you find the time at some point in the future it might be a good idea to update this module to extend the core `ResourceTracker` module (it would be significantly less work to maintain then since it automatically generates all the spenders and generators without you having to define them).

Reply:
I'll try to look into that (for a future PR to not bloat this one even more), might not be a lot of problem, but I can see some troubles in Destruction, I think that one *still* hasn't got energize events for the Soul Shard Fragments (but I haven't explicitly looked into the logs).